ouroboros-consensus/changelog.d/20241128_084625_georgy.lukyanov_892_checksum_snaphot_file.md

```diff
@@ -0,0 +1,12 @@
+### Breaking
+
+- When writing a ledger state snapshot to disk, calculate the state's CRC checksum and write it to a separate file, which is named the same as the snapshot file, plus the `.checksum` extension.
+- When reading a snapshot file, calculate its checksum and compare it to the value in the corresponding `.checksum` file. Return an error if either the checksum file does not exist or the checksum is different or invalid.
```

Comment:
Perhaps if the file was missing we could emit a warning but not fail 🤔

Reply:
Yes, great idea. I think that is actually necessary, as otherwise we'd need to delete the snapshots or create the checksums manually when starting an updated node on top of the old node's state. I'll make this change.

Reply:
The solution I've chosen: in `initLedgerDB`, we catch the error if the `.checksum` file does not exist, issue a warning and retry reading the snapshot without the checksum file.